src/common/partition_builder.h

```diff
@@ -218,7 +232,27 @@ class PartitionBuilder {
         }
       }
     } else {
-      LOG(FATAL) << "Column data split is only supported for the `approx` tree method";
+      auto pred_hist = [&](auto ridx, auto bin_id) {
```

Comment:
Logically, is there any difference from https://github.com/dmlc/xgboost/blob/07b2d5a26d96888fa4cec4dd1b1ef0b31d0c3fbb/src/common/partition_builder.h#L126 ?

Reply:
Yes they are the same, but since it's a closure, can't be easily reused. Maybe we should extract into a standalone function?

Reply:
The partitioner is becoming less and less manageable now. I don't have any good suggestions on how to refactor it at the moment and can approve the PR as-is. But would be great if there's a way to simplify/modularize it.

Reply:
Would it be helpful to separate out the column-split logic into its own class?

Reply:
Maybe next time. Currently, we have total 8 combinations for `tree_method, categorical, column-split`, along with these, we also need to take multi-output into account. I'm trying to think of a way to make the code easier to understand and write a test suite specifically for partition builder.